# Task 5 (Advanced) – Content Rating & Genre Analysis
**Goal:** Examine Netflix content ratings and genres to understand audience targeting.

**Workflow:** Analyze ratings → Identify top genres → Compare ratings across content types → Build interactive visualizations → Present audience insights.

In [1]:
import pandas as pd
import plotly.express as px

import os
path = "/content/Dataset.csv" if os.path.exists("/content/Dataset.csv") else "Dataset.csv"
df = pd.read_csv(path)

In [2]:
"""
Step 1: Analyze rating categories.

I used value_counts() to calculate the number of titles in each rating
category. This helps identify the most common content ratings and provides
an initial understanding of the audience categories represented in the
Netflix dataset.
"""

'\nStep 1: Analyze rating categories.\n\nI used value_counts() to calculate the number of titles in each rating\ncategory. This helps identify the most common content ratings and provides\nan initial understanding of the audience categories represented in the\nNetflix dataset.\n'

## Step 1: Analyze Rating Categories
Count titles in each rating category to understand the audience segments Netflix content targets.

In [3]:
rating_counts = df["rating"].value_counts()
print(rating_counts)
#TV-MA is the most common rating, indicating that a large portion of the content in the dataset is targeted toward mature audiences.

rating
TV-MA       3205
TV-14       2157
TV-PG        861
R            799
PG-13        490
TV-Y7        333
TV-Y         306
PG           287
TV-G         220
NR            79
G             41
TV-Y7-FV       6
NC-17          3
UR             3
Name: count, dtype: int64


In [4]:
"""
Step 2: Identify the most common genres.

The listed_in column can contain multiple genres for a single title.
I split these genre combinations into individual categories using split()
and explode(), then counted their frequency using value_counts().

This allows me to identify the most common genres in the Netflix dataset.
"""

'\nStep 2: Identify the most common genres.\n\nThe listed_in column can contain multiple genres for a single title.\nI split these genre combinations into individual categories using split()\nand explode(), then counted their frequency using value_counts().\n\nThis allows me to identify the most common genres in the Netflix dataset.\n'

## Step 2: Identify the Most Common Genres
The `listed_in` column holds multiple genres per title, so we split and explode it into individual genre rows before counting frequency.

In [5]:
genres = df["listed_in"].str.split(", ").explode()
print(genres)

0                Documentaries
1               Crime TV Shows
1       International TV Shows
1        TV Action & Adventure
2                    TV Dramas
                 ...          
8787    International TV Shows
8787         Romantic TV Shows
8787                 TV Dramas
8788                  Kids' TV
8789                  Kids' TV
Name: listed_in, Length: 19294, dtype: str


In [6]:
genre_counts = genres.value_counts()
top_genres = genre_counts.head(10).reset_index()
print(top_genres)

                  listed_in  count
0      International Movies   2752
1                    Dramas   2426
2                  Comedies   1674
3    International TV Shows   1349
4             Documentaries    869
5        Action & Adventure    859
6                 TV Dramas    762
7        Independent Movies    756
8  Children & Family Movies    641
9           Romantic Movies    616


In [7]:
"""
Step 3: Compare ratings across content types.

I used a cross-tabulation to compare rating categories between Movies and
TV Shows. This allows me to examine how the distribution of content ratings
differs between the two content types.
"""

'\nStep 3: Compare ratings across content types.\n\nI used a cross-tabulation to compare rating categories between Movies and\nTV Shows. This allows me to examine how the distribution of content ratings\ndiffers between the two content types.\n'

## Step 3: Compare Ratings Across Content Types
Cross-tabulate rating vs. type to see how the rating mix differs between Movies and TV Shows.

In [8]:
rating_by_type = pd.crosstab(df["rating"], df["type"])
print(rating_by_type)

type      Movie  TV Show
rating                  
G            41        0
NC-17         3        0
NR           75        4
PG          287        0
PG-13       490        0
R           797        2
TV-14      1427      730
TV-G        126       94
TV-MA      2062     1143
TV-PG       540      321
TV-Y        131      175
TV-Y7       139      194
TV-Y7-FV      5        1
UR            3        0


In [9]:
# Step 4 — Interactive visualizations

## Step 4: Interactive Visualizations
Build interactive Plotly charts for rating distribution, top genres, and ratings by content type.

In [10]:
rating_df = rating_counts.reset_index()

rating_df.columns = ["rating", "content_count"]

fig = px.bar(
    rating_df,
    x="rating",
    y="content_count",
    title="Netflix Content Rating Distribution",
    labels={
        "rating": "Rating",
        "content_count": "Number of Titles"
    }
)

fig.show()

In [11]:
# Top Genres
top_genres = genre_counts.head(10).reset_index()

top_genres.columns = ["genre", "content_count"]

fig = px.bar(
    top_genres,
    x="content_count",
    y="genre",
    orientation="h",
    title="Top 10 Netflix Genres",
    labels={
        "genre": "Genre",
        "content_count": "Number of Titles"
    }
)

fig.show()

In [12]:
# Ratings by Content Type

rating_type_df = rating_by_type.reset_index()
print(rating_type_df)

fig = px.bar(
    rating_type_df,
    x="rating",
    y=["Movie", "TV Show"],
    barmode="group",
    title="Netflix Rating Distribution by Content Type",
    labels={
        "rating": "Rating",
        "value": "Number of Titles",
        "variable": "Content Type"
         }
)
fig.show()


type    rating  Movie  TV Show
0            G     41        0
1        NC-17      3        0
2           NR     75        4
3           PG    287        0
4        PG-13    490        0
5            R    797        2
6        TV-14   1427      730
7         TV-G    126       94
8        TV-MA   2062     1143
9        TV-PG    540      321
10        TV-Y    131      175
11       TV-Y7    139      194
12    TV-Y7-FV      5        1
13          UR      3        0


In [13]:
rating_percentage = (
    df["rating"]
    .value_counts(normalize=True)
    .mul(100)
    .round(2)
)

print(rating_percentage)
#TV-MA represents 36.46% of the content.

rating
TV-MA       36.46
TV-14       24.54
TV-PG        9.80
R            9.09
PG-13        5.57
TV-Y7        3.79
TV-Y         3.48
PG           3.27
TV-G         2.50
NR           0.90
G            0.47
TV-Y7-FV     0.07
NC-17        0.03
UR           0.03
Name: proportion, dtype: float64


### Rating & Genre Share (%)
Express ratings and genres as percentages of the total catalog for easier comparison.

In [14]:
genre_counts.head(10)

listed_in
International Movies        2752
Dramas                      2426
Comedies                    1674
International TV Shows      1349
Documentaries                869
Action & Adventure           859
TV Dramas                    762
Independent Movies           756
Children & Family Movies     641
Romantic Movies              616
Name: count, dtype: int64

In [15]:
genre_percentage = (
    genre_counts
    .div(len(df))
    .mul(100)
    .round(2)
)

print(genre_percentage.head(10))

listed_in
International Movies        31.31
Dramas                      27.60
Comedies                    19.04
International TV Shows      15.35
Documentaries                9.89
Action & Adventure           9.77
TV Dramas                    8.67
Independent Movies           8.60
Children & Family Movies     7.29
Romantic Movies              7.01
Name: count, dtype: float64


In [16]:
rating_by_type_percentage = pd.crosstab(
    df["rating"],
    df["type"],
    normalize="columns"
).mul(100).round(2)

print(rating_by_type_percentage)
#TV-MA represents  33.66% of TV Shows compared with  42.91% of Movies.

type      Movie  TV Show
rating                  
G          0.67     0.00
NC-17      0.05     0.00
NR         1.22     0.15
PG         4.68     0.00
PG-13      8.00     0.00
R         13.01     0.08
TV-14     23.29    27.40
TV-G       2.06     3.53
TV-MA     33.66    42.91
TV-PG      8.81    12.05
TV-Y       2.14     6.57
TV-Y7      2.27     7.28
TV-Y7-FV   0.08     0.04
UR         0.05     0.00


## Step 5: Present Audience Insights
Pull together the rating and genre figures computed above into a written summary.

In [17]:
# STEP 5 : Audience Insights

In [18]:
print(rating_percentage.head(10))

rating
TV-MA    36.46
TV-14    24.54
TV-PG     9.80
R         9.09
PG-13     5.57
TV-Y7     3.79
TV-Y      3.48
PG        3.27
TV-G      2.50
NR        0.90
Name: proportion, dtype: float64


In [19]:
print(genre_counts.head(10))

listed_in
International Movies        2752
Dramas                      2426
Comedies                    1674
International TV Shows      1349
Documentaries                869
Action & Adventure           859
TV Dramas                    762
Independent Movies           756
Children & Family Movies     641
Romantic Movies              616
Name: count, dtype: int64


In [20]:
print(rating_by_type_percentage)

type      Movie  TV Show
rating                  
G          0.67     0.00
NC-17      0.05     0.00
NR         1.22     0.15
PG         4.68     0.00
PG-13      8.00     0.00
R         13.01     0.08
TV-14     23.29    27.40
TV-G       2.06     3.53
TV-MA     33.66    42.91
TV-PG      8.81    12.05
TV-Y       2.14     6.57
TV-Y7      2.27     7.28
TV-Y7-FV   0.08     0.04
UR         0.05     0.00


## Audience Insights — Executive Summary

### Rating Analysis
**TV-MA is the single most common rating**, covering **36.46%** of all titles (3,205 of 8,790), followed by **TV-14 (24.54%, 2,157 titles)**. Together these two mature-audience ratings account for **over 60% of the entire catalog**. Family/general-audience ratings — TV-G (2.50%), G (0.47%), TV-Y (3.48%) — make up only a small fraction, confirming that adult and teen viewers are Netflix's primary audience segments rather than young children.

### Genre Analysis
The top genres are **International Movies (2,752 titles, 31.3%)**, **Dramas (2,426, 27.6%)**, **Comedies (1,674, 19.0%)**, **International TV Shows (1,349, 15.4%)**, and **Documentaries (869, 9.9%)**. The strong showing of "International" categories at #1 for both Movies and TV Shows reflects a deliberate global-content sourcing strategy, while Dramas and Comedies anchor the platform's mainstream appeal.

### Content Type and Rating Comparison
The rating mix differs meaningfully by content type: **TV-MA represents 42.91% of TV Shows but only 33.66% of Movies**. In other words, TV Shows skew more heavily toward mature content than Movies do — likely because long-running series (crime dramas, thrillers) more often carry mature ratings, while the movie catalog includes a wider mix of family and general-audience titles alongside its mature-rated films.

### Overall Audience Insight
Netflix's content strategy leans toward **mature, internationally-sourced drama and comedy content**, with TV Shows skewing more mature than Movies. This points to a core audience of teens and adults, a strong emphasis on international storytelling, and a supporting (but smaller) family-content offering. For content strategy, this suggests continued investment in mature dramas/comedies and international titles will resonate with the platform's core viewer base, while family-rated content remains a comparatively underweighted category worth monitoring for growth opportunities.